tensorpack/cmtf.py

```diff
@@ -226,19 +226,24 @@ def perform_CP(tOrig, r=6, tol=1e-6, maxiter=50, progress=False):
 
     R2X_last = -np.inf
     tFac.R2X = calcR2X(tFac, tOrig)
+    if track_error:
```

Comment:
All that should happen here is if `callback` is not None, it gets called with various internal information about the fitting process. For example, `callback(tFac)`.